src/fortranapex/apexsh.f90

```diff
@@ -150,86 +151,88 @@ module apxshmodule
     real(8)                  :: xqgrad(1:3), yqgrad(1:3), zqgrad(1:3)
     real(8)                  :: qlatgrad(1:3), qlongrad(1:3)
 
-    real(8), parameter       :: pi=3.14159265358979323846D0
-    real(8), parameter       :: dtor=pi/180D0, pid2=pi/2D0, twopi=2D0*pi
-    real(8), parameter       :: Req=6378.1370D0, eps=1.D0/298.257223563D0
-    real(8), parameter       :: Re=Req*(1-eps/3D0), ecc2=eps*(2-eps)
-    real(4), parameter       :: missing=-9999E0
+    ! Req = Equatorial radius of Earth in km (WGS84 value)
+    ! eps = flatness of ellipsoidal Earth (WGS84 value)
+    ! Re = Mean radius of Earth in km
+    ! ecc2 = squared eccentricity of ellipsoidal Earth
 
-!   Req = Equatorial radius of Earth in km (WGS84 value)
-!   eps = flatness of ellipsoidal Earth (WGS84 value)
-!   Re = Mean radius of Earth in km
-!   ecc2 = squared eccentricity of ellipsoidal Earth
+    real(8), parameter       :: pi=3.14159265358979323846D0
+    real(8), parameter       :: dtor=pi / 180D0, pid2=pi / 2D0, twopi=2D0 * pi
+    real(8), parameter       :: Req=6378.1370D0, eps=1.D0 / 298.257223563D0
+    real(8), parameter       :: Re=Req * (1 - eps / 3D0), ecc2=eps * (2 - eps)
```

Comment:
Perhaps it would be safer to assign these as constant values instead of calculations?  Saying this specifically because of the `ecc2` variable in the fortranapex.pyf file.

Reply:
I think this might be related to the constants not being declared explicitly as doubles.  I have revised the block as follows:
```
    real(8), parameter       :: pi=3.14159265358979323846D0
    real(8), parameter       :: dtor=pi / 180D0, pid2=pi / 2D0, twopi=2D0 * pi
    real(8), parameter       :: Req=6378.1370D0, eps=1D0 / 298.257223563D0
    real(8), parameter       :: Re=Req * (1D0 - eps / 3D0)
    real(8), parameter       :: ecc2=eps * (2D0 - eps)
    real(8), parameter       :: missing=- 9999E0
```

Reply:
Potentially unrelated, but there are similar blocks that define modules in `apex.f90` (line 7) and `magfld.f90` (line 23).  Should they all be combined in some way?

Reply:
That would be a lot cleaner, if it's not a big deal, go ahead and make the change.

Reply:
I made this change, all these coefficients are now defined in the `coeffmodule` in `magfld.f90`, and imported to other modules from there.  One interesting point is the comments in `apex.f90` talk extensively about how eccentricity is defined, however the actual code seems to do the opposite of what the comments say.  I tried changing it to what the comments say, but this makes a bunch of the unit tests fail (I think floating point errors again).  Obviously we can change this if we want, but we'll have to change the unit tests again.